[Documentation] Parameters & Authorizations (#1182)
pokt-network/poktroll: docusaurus/docs/operate/upgrades/3_module_auth.md

---
title: Module Authorizations
sidebar_position: 3
---

## Module Authorizations <!-- omit in toc -->

> Pocket Network utilizes an offchain governance mechanism that enables the community to vote on proposals.

Reviewer: ```suggestion
Pocket Network utilizes an onchain governance mechanism that enables the community to vote on proposals.
```

> Once a proposal passes, or a decision by PNF is made on behalf of the DAO, PNF adjust the parameters necessary for the protocol's operation.

Reviewer: Currently:
```suggestion
Once a proposal passes, or a decision by PNF is made on behalf of the DAO, the parameters updates are applied.
```

As soon as #1154 is merge, the following statement will be true:
```suggestion
Once a proposal passes, or a decision by PNF is made on behalf of the DAO, the parameter updates are pending activation, onchain, and active starting from the beginning of the next session.
```

- [Access Control](#access-control)
  - [MainNet Authorizations](#mainnet-authorizations)
    - [`x/gov` Module Granter](#xgov-module-granter)
    - [`PNF` Account Grantee](#pnf-account-grantee)
- [Examples](#examples)
  - [Example: Adding a new Authorization](#example-adding-a-new-authorization)

## Access Control

### MainNet Authorizations

The list of authorizations enabled on MainNet genesis can be found at [pokt-network/pocket-network-genesis/tree/master/shannon/mainnet](https://github.com/pokt-network/pocket-network-genesis/tree/master/shannon/mainnet).

> #### `x/gov` Module Granter

Reviewer: ```suggestion
#### `x/gov` Module Account
```

> The `x/gov` module granter is tied to address `pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t`.

Reviewer: ```suggestion
The `x/gov` module account is deterministically tied to address `pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t`. This will be true for any "pocket" network; e.g., LocalNet, TestNet, MainNet, etc.
```

Not sure how much detail you think is relevant but in case you think it's worth clarifying the relevance of the `x/gov` module account - why is it part of this picture? - may I suggest the following additional context for your consideration:

The `x/gov` module account is the default configured "authority" for all cosmos-sdk modules. Each module references its own configured "authority" when executing messages which require authorization (e.g. `MsgUpdateParams` messages).

No one has access to this address, but the grants it has provided to other accounts can be queried like so:

```bash
pocketd query authz grants-by-granter pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t --node https://shannon-grove-rpc.mainnet.poktroll.com
```

#### `PNF` Account Grantee

> The grants the `x/gov` module granter has provided to the `PNF` account can be queried like so:

Reviewer: ```suggestion
The authorizations which the `x/gov` module account has granted to the `PNF` account can be queried like so:
```
